packages/native/src/useScrollToTop.tsx

```diff
@@ -49,53 +49,62 @@ export default function useScrollToTop(
   const route = useRoute();
 
   React.useEffect(() => {
-    let current = navigation;
-
-    // The screen might be inside another navigator such as stack nested in tabs
-    // We need to find the closest tab navigator and add the listener there
-    while (current && current.getState().type !== 'tab') {
-      current = current.getParent();
+    let tabNavigators = [];
+    let currentNavigator = navigation;
+    
+     // The screen might be inside another navigator such as stack nested in tabs
+     // The screen might be in multiple tab bars (as a top tab bar and a bottom tab bar and we want both events to work)
+     // We need to find the closest tab navigators and add the listener there
+     while (currentNavigator) {
+      if(currentNavigator.getState().type === 'tab') {
+        tabNavigators = tabNavigators.concat(currentNavigator);
```

Comment:
mutation is fine for local variables
```suggestion
        tabNavigators.push(currentNavigator);
```